support/node-health-check/src/main.rs

```diff
@@ -0,0 +1,107 @@
+use std::convert::Infallible;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::{sync::Arc, time::Duration};
+
+use anyhow::Result;
+use reqwest::StatusCode;
+use serde::{Deserialize, Serialize};
+use structopt::StructOpt;
+use warp::Filter;
+
+/// Health struct returned by the RPC
+#[derive(Debug, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Health {
+    /// Number of connected peers
+    pub peers: usize,
+    /// Is the node syncing
+    pub is_syncing: bool,
+    /// Should this node have any peers
+    /// Might be false for local chains or when running without discovery.
+    pub should_have_peers: bool,
+}
+
+#[derive(Debug, StructOpt, Clone)]
+#[structopt(
+    name = "node-health-check",
+    about = "Sidecar for substrate node to report health compatible with AWS ALB."
+)]
+struct Opt {
+    /// Debug
+    #[structopt(short, long)]
+    debug: bool,
+
+    /// Set poll_interval
+    #[structopt(short = "i", default_value = "5")]
+    poll_interval: u64,
+
+    /// Set minimum peers
+    #[structopt(default_value = "2")]
+    min_peers: usize,
+
+    /// Node rpc endpoint
+    #[structopt(short, long, env, default_value = "http://localhost:9933/health")]
+    node_rpc_endpoint: String,
+
+    // The port to listen on
+    #[structopt(short, long, default_value = "9955")]
+    port: u16,
+}
+
+async fn poller(opts: Opt, healthy: Arc<AtomicBool>) {
+    loop {
+        let resp = async {
+            reqwest::get(&opts.node_rpc_endpoint)
+                .await?
+                .json::<Health>()
+                .await
+        };
+
+        match resp.await {
+            Ok(health) => {
+                let node_is_healthy = !health.is_syncing
+                    && (!health.should_have_peers || health.peers >= opts.min_peers);
```

Comment:
Move this into a method on Health. Then you can write it more readably:

```rust
fn is_healthy(&self, required_peers: usize) -> bool {
  if self.is_syncing {
    return false;
  }

  if self.should_have_peers {
    return self.peers >= required_peers;
  }

  return true;
}
```

Reply:
Yes, agree, this expresses the logic more readable. Instead of the boolean logic.